src/utils/codec.ts

```
@@ -541,6 +748,53 @@ function assertIndexArrayInRange(arr: readonly unknown[], bound: number, label:
   }
 }
 
+function assertSparseDataDomain(
+  data: readonly unknown[],
+  dtype: string,
+  shape: readonly number[]
+): void {
+  const expectsBoolean = dtype === 'bool' || dtype === 'bool_';
+  const integerMatch = /^(u?int)(\d*)$/.exec(dtype);
```

ESLint

[error] Move this regular expression to module scope to avoid re-compilation on every call.

(e18e/prefer-static-regex)